src/util/dstring.h

```diff
@@ -123,6 +123,16 @@ class dstringt
     return out << as_string();
   }
 
+  friend size_t hash_string(const dstring &s)
+  {
+    return s.hash();
+  }
+
+  friend size_t const_hash_string(const dstring &s)
+  {
+    return hash_string(string_container.get_string(s.no));
+  }
+
```

Comment:
In line with what @martin-cs said: what exactly do these seek to achieve? In particular, the latter actually invokes the much more costly `std::string` version of `hash_string` !?